systemtest/src/main/java/io/strimzi/systemtest/VertxFactory.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+
+/**
+ * Creates minimal vertx instances
+ */
+public class VertxFactory {
+    public static Vertx create() {
+        VertxOptions options = new VertxOptions()
+                .setWorkerPoolSize(1)
+                .setInternalBlockingPoolSize(1)
+                .setEventLoopPoolSize(1);
```

Comment:
I take over this part from enmaase. They used this in dark times :), where they had issues with vert.x and they wanted minimal instance. Definitely I can try to rewrite it.

Reply:
I was just curious, maybe they wanted to enforce these limits just to avoid doing something wrong during the tests. Of course, I don't know the real reason btw :-)

Reply:
> Vertex recommends us to use values in 20 times bigger than we have.

That makes no sense, and I think it might demonstrate a misconception.

Vertx is about providing async APIs. But a lot of Java's APIs are blocking. So to compensate for this (for those APIs which vertx itself needs to use internally), it will need to use a pool of threads to make those blocking calls. That's what this internal blocking pool is. It doesn't matter how many processors/cores you have, because things using that pool of threads will be blocked most of the time. The OS will schedule other threads to run instead of those blocked threads. So even on a machine with 1 core `internalBlockingPoolSize = 20` would be _fine_. Those 20 threads would each take turns on the processor, block and eventually be runnable again when the network socket, file operation or whatever had completed.

The workerPoolSize is just the defailt for a user (i.e. non-internal) blocking pool.

So in vertx all the blocking calls are happening from threads which are in those pools. The event loop runs "all the rest of the code". Again it's not a problem having more event loops than physical cores because the OS will schedule threads.